### DISCUSSION TITLE
Running .NET 7 single file binary on Apple M1 chip causes app to be "killed"

### DISCUSSION THREAD

**Discussion by rcdailey** [dotnet/runtime#79229] [Q&A]
I have a Github Workflow that compiles my .NET 7 console application using a windows agent. It runs this in a powershell script:

```pwsh
dotnet publish MyProject.csproj `
    --output publish `
    --configuration Release`
    --runtime osx-arm64 `
    --self-contained true `
    -p:PublishSingleFile=true `
    -p:IncludeNativeLibrariesForSelfExtract=true `
    -p:PublishReadyToRunComposite=true `
    -p:PublishReadyToRunShowWarnings=true `
    -p:EnableCompressionInSingleFile=true 
```

This is the platform I'm running on:

[![enter image description here][1]][1]

My application is not signed, but Mac lets me bypass that for testing purposes. Eventually it just fails like so:

```
[1]    37470 killed     ./myconsoleapp --help
```

I get no further diagnostics. What steps can I take to further diagnose the issue? Why is `osx-arm64` not working but `osx-x64` works just fine?

  [1]: https://i.stack.imgur.com/gdITN.png

**Comment by am11** [comment]
Does setting `EnableCompressionInSingleFile` to `false` fix the error?

**Reply by am11** [reply]
@VSadov, @vitek-karas. this seems related to https://github.com/dotnet/runtime/pull/49855.
I get `bus error (138)` on osx-arm64 even with hello-world app, e.g. using daily builds:
```sh
$ dotnet8 publish -o publish -p:PublishSingleFile=true -p:EnableCompressionInSingleFile=true
$ ./publish/helloworldapp # bus error

$ dotnet8 publish -o publish -p:PublishSingleFile=true -p:EnableCompressionInSingleFile=false
$ ./publish/helloworldapp # works
```

**Reply by vitek-karas** [reply]
I created an issue from this: https://github.com/dotnet/runtime/issues/79267
